src/Disciplina/Educator/Validate.hs

```diff
@@ -0,0 +1,41 @@
+
+-- | Validation of private block
+
+module Disciplina.Educator.Validate
+       ( validate
+       ) where
+
+import Universum
+
+import Control.Monad.Except (MonadError)
+import Serokell.Util.Verify (verifyGeneric, verResToMonadError)
+
+import Disciplina.Educator.Block (PrivateBlock (..), PrivateBlockHeader (..), pbBody, pbbTxs)
+import Disciplina.Educator.Txs (PrivateTx (..), PrivateTxAux (..), PrivateTxWitness (..))
+import Disciplina.Educator.Serialise ()
+import Disciplina.Core (Submission (..), SignedSubmission (..), Address (..))
+import Disciplina.Crypto (fromFoldable, getMerkleRoot, hash, verify, )
+
+-- | Validate private block.
+-- Block is valid if
+-- 1. Header matches body
+-- 2. All transactions have valid signatures
+-- 3. Transaction signature public key correspond to student public key
+validate :: (MonadError [Text] m) => PrivateBlock -> m ()
```

Comment:
This is fully pure code which isn't meant to be run in any IO-based monad, so it's better to simply use `Either [Text] ()` as a return type here. See `exceptions` guide: https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md#pure-code-regular-errors

Reply:
Also, naming the function simply `validate` isn't nice, be more specific, e.g. `validatePrivateBlock`.